esp-bootloader-esp-idf/src/lib.rs

```diff
@@ -0,0 +1,261 @@
+        mmu_page_size: u32,
+    ) -> Self {
+        Self {
+            magic_word: ESP_APP_DESC_MAGIC_WORD,
+            secure_version: 0,
+            reserv1: [0; 2],
+            version: str_to_cstr_array(version),
+            project_name: str_to_cstr_array(project_name),
+            time: str_to_cstr_array(build_time),
+            date: str_to_cstr_array(build_date),
+            idf_ver: str_to_cstr_array(idf_ver),
+            app_elf_sha256: [0; 32],
+            min_efuse_blk_rev_full,
+            max_efuse_blk_rev_full,
+            mmu_page_size: (mmu_page_size.ilog2()) as u8,
+            reserv3: [0; 3],
+            reserv2: [0; 18],
+        }
+    }
+
+    /// The magic word - should be `0xABCD5432`
+    pub fn magic_word(&self) -> u32 {
+        self.magic_word
+    }
+
+    /// Secure version
+    pub fn secure_version(&self) -> u32 {
+        self.secure_version
+    }
+
+    /// Application version
+    pub fn version(&self) -> &str {
+        array_to_str(&self.version)
+    }
+
+    /// Application name
+    pub fn project_name(&self) -> &str {
+        array_to_str(&self.project_name)
+    }
+
+    /// Compile time
+    pub fn time(&self) -> &str {
+        array_to_str(&self.time)
+    }
+
+    /// Compile data
+    pub fn date(&self) -> &str {
+        array_to_str(&self.date)
+    }
+
+    /// IDF version
+    pub fn idf_ver(&self) -> &str {
+        array_to_str(&self.idf_ver)
+    }
+
+    /// SHA256
+    ///
+    /// The default tooling won't populate this
+    pub fn app_elf_sha256(&self) -> &[u8; 32] {
+        &self.app_elf_sha256
+    }
+
+    /// Minimal eFuse block revision supported by image
+    ///
+    /// Format `major * 100 + minor`
+    pub fn min_efuse_blk_rev_full(&self) -> u16 {
+        self.min_efuse_blk_rev_full
+    }
+
+    /// Maximal eFuse block revision supported by image
+    ///
+    /// Format `major * 100 + minor`
+    pub fn max_efuse_blk_rev_full(&self) -> u16 {
+        self.max_efuse_blk_rev_full
+    }
+
+    /// MMU page size in bytes
+    pub fn mmu_page_size(&self) -> u32 {
+        (2 as u32).pow(self.mmu_page_size as u32)
+    }
+}
+
+impl core::fmt::Debug for EspAppDesc {
+    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+        f.debug_struct("EspAppDesc")
+            .field("magic_word", &self.magic_word)
+            .field("secure_version", &self.secure_version)
+            .field("version", &self.version())
+            .field("project_name", &self.project_name())
+            .field("time", &self.time())
+            .field("date", &self.date())
+            .field("idf_ver", &self.idf_ver())
+            .field("app_elf_sha256", &self.app_elf_sha256)
+            .field("min_efuse_blk_rev_full", &self.min_efuse_blk_rev_full)
+            .field("max_efuse_blk_rev_full", &self.max_efuse_blk_rev_full)
+            .field(
+                "mmu_page_size",
+                &((2 as u32).pow(self.mmu_page_size as u32)),
```

Comment:
```suggestion
                &self.mmu_page_size(),
```

Could we deduplicate, please?

Reply:
you mean debug should just show the real value - I agree

Reply:
No, I meant we shouldn't inline the contents of `fn mmu_page_size`, especially that it's not simple.

Reply:
ah - mhhh I'm not sure if we should show the calculated value in debug - I thought we should but after thinking about it , maybe not

Reply:
I'm okay with that, I think.

Could you also add defmt::Format, please?